### PR TITLE
Document config variables and rename node to partner

### LINKED ISSUE #110: Document the different parameters available in config.yml

### PR DESCRIPTION
**Pull request by bowni**
Closes #110 and #109

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,24 +86,118 @@ The computation of the Shapley Values quickly becomes intensive when the number
...
+Percentages of the original dataset each partner receives to mock a collaborative ML scenario where each partner provides data for the ML training.  
+Example: `amounts_per_partner: [0.3, 0.3, 0.1, 0.3]`
+
+`samples_split_option`: `random` or `stratified`  
+How the original dataset data samples are split among partners:
+
+- `random`: the dataset is shuffled and the split is done randomly, ensuring a homogeneous data distribution among partners
+- `stratified`: the dataset is stratified per labels, so that partners have datasets covering different regions of space
+
+`corrupted_partners`: `[not_corrupted (default), shuffled or corrupted]`
```
Comment (RomainGoussault) [inline README.md:139]: `corrupted_partners` makes me believe that one of the partner belongs to the mafia :).
Suggestions: corrupted_data ? data_corruption_status ? data_quality_per_partner ?

File: README.md [hunk 2]
```diff
@@ -68,24 +86,118 @@ The computation of the Shapley Values quickly becomes intensive when the number
...
+  
+Example: `[not_corrupted, not_corrupted, not_corrupted, shuffled]`
+
+##### Configuration of the distributed learning approach
+
+`aggregation_weighting`: `uniform` (default), `data_volume` or `local_score`  
+After a training iteration over a given mini-batch, how individual models of each partner are aggregated:
+
+- `uniform`: simple average (non-weighted)
+- `data_volume`: average weighted with the relative amounts of data (in number of data samples)
```
Comment (RomainGoussault) [inline README.md:154]: (in number of data samples) --> (number of data samples)

File: README.md [hunk 3]
```diff
@@ -68,24 +86,118 @@ The computation of the Shapley Values quickly becomes intensive when the number
...
+
+`epoch_count`: `int` (default: `40`)  
+Number of epochs passed as argument in the `.fit()` function. Superseded when `is_early_stopping` is set to `true`.  
+Example: `epoch_count: 30`
+
+`minibatch_count`: `int` (default: `20`)  
+Number of mini batches into which each partner's dataset is split, and over which are performed the iterations of parallel local training plus aggregation.  
+Example: `minibatch_count: 20`
+
+`is_early_stopping`: `True` (default) or `False`
```
Comment (RomainGoussault) [inline README.md:168]: when the performance **on the validation set** reaches ...

File: README.md [hunk 4]
```diff
@@ -68,24 +86,118 @@ The computation of the Shapley Values quickly becomes intensive when the number
...
+- `local_score`: average weighted with the performance (on a central validation set) of the individual models
+ 
+`single_partner_test_mode`: `global` (default) or `local`  
+When training a model on a single partner (this is needed in certain contributivity measurement approaches), defines if the final performance is tested on the central testset or on the partner's local testset. Note: a train-test split is performed on the original dataset, forming a central testset; this testset is also split over each partner (randomly) forming local testsets.  
+
+`epoch_count`: `int` (default: `40`)  
+Number of epochs passed as argument in the `.fit()` function. Superseded when `is_early_stopping` is set to `true`.  
+Example: `epoch_count: 30`
+
+`minibatch_count`: `int` (default: `20`)
```
Comment (RomainGoussault) [inline README.md:164]: > and over which are performed the iterations of parallel local training plus aggregation.  
This bit is not crystal clear.

File: README.md [hunk 5]
```diff
@@ -68,24 +86,118 @@ The computation of the Shapley Values quickly becomes intensive when the number
...
+After a training iteration over a given mini-batch, how individual models of each partner are aggregated:
+
+- `uniform`: simple average (non-weighted)
+- `data_volume`: average weighted with the relative amounts of data (in number of data samples)
+- `local_score`: average weighted with the performance (on a central validation set) of the individual models
+ 
+`single_partner_test_mode`: `global` (default) or `local`  
+When training a model on a single partner (this is needed in certain contributivity measurement approaches), defines if the final performance is tested on the central testset or on the partner's local testset. Note: a train-test split is performed on the original dataset, forming a central testset; this testset is also split over each partner (randomly) forming local testsets.  
+
+`epoch_count`: `int` (default: `40`)
```
Comment (RomainGoussault) [inline README.md:160]: Should we replace `epoch_count` by the more appropriate `federated_learning_rounds `?

**Comment by bowni** [comment]
Ok, great comments thanks @RomainGoussault. I am on it, so hopefully we can start tomorrow's bootcamp with this behind us, let's try!

File: README.md [hunk 6]
```diff
@@ -68,24 +86,118 @@ The computation of the Shapley Values quickly becomes intensive when the number
...
+#### Scenario-level parameters
+
+##### Definition of collaborative scenarios
+
+`partners_count`: `int`  
+Number of partners in the mocked collaborative ML scenario.  
+Example: `partners_count: 4`
+
+`amounts_per_partner`: `[float]`  
+Percentages of the original dataset each partner receives to mock a collaborative ML scenario where each partner provides data for the ML training.
```
[the remark this answers is not in the document]
Comment (bowni) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (bowni) [reply README.md]: Done, renamed as `corrupted_datasets`

File: README.md (hunk 2 above)
Comment (bowni) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (bowni) [reply README.md:168]: Done

File: README.md (hunk 4 above)
Comment (bowni) [reply README.md:164]: Clarified

File: README.md (hunk 5 above)
Comment (bowni) [reply README.md:160]: I think I'd prefer to keep it that way. This is really close to what an epoch is. And this makes quite clear for people not familiar with distributed learning that this is the same concept (an entire pass over the dataset, that has to be done multiple times).

**Pull request merged**